### PR TITLE
feat: shared session UI affordance — server contract (Part 2a of #678) (closes #774)

### LINKED ISSUE #774: feat: shared session UI affordance — server contract (Part 2a of #678)
## Scope (Part 2a of #678)

Implement the **server contract** for shared session UI affordance — the data layer that Part 2b (client UI, separate sub-issue) consumes.

Per [`docs/design/shared-orchestrator-session.md` §UI](../blob/main/docs/design/shared-orchestrator-session.md):

- Add `sharedAccountsAvailable: boolean` to `/api/config` (provisional shape; future evolution: `{id, displayName}[]` once multi-account configuration lands).
- Add `isShared: boolean` derived field to session payload, computed server-side from `sessions.created_by` resolving to a configured shared account.
- Existing `Session.createdBy` field is preserved as-is. Multi-user UX requires per-session owner identification (existing `mine` filter, future #683 tree view, future #684 owner display).

## Out of scope (covered by Part 2b — separate sub-issue, blocked by this)

- "Create shared session" button
- Visual distinction (badge / colour / label) in session list
- Filter `shared` option

## Hypothesized implementation layer (high confidence)

- `/api/config` route: `packages/server/src/routes/api.ts:25-36` (existing endpoint, extend the response object).
- `SharedAccountRegistry`: `packages/server/src/services/shared-account-registry.ts` already provides `isSharedUserId(userId)` (PR #764, Slice 1). Reuse — do NOT add a parallel mechanism.
- Session payload serializer: trace from session route response back through `session-converter-service.ts` / `session-manager.ts`; extend the serialized shape rather than adding a new endpoint.
- Shared types: `packages/shared/src/types/session.ts` and `packages/shared/src/schemas/session.ts` for `isShared`.

### Functional

- [ ] `/api/config` returns `sharedAccountsAvailable: false` when `AGENT_CONSOLE_SHARED_USERNAME` is unset → server unit test
- [ ] `/api/config` returns `sharedAccountsAvailable: true` when `AGENT_CONSOLE_SHARED_USERNAME` resolves to an existing OS account (via `SharedAccountRegistry`) → server unit test
- [ ] Session payload `isShared` is `true` for sessions whose `created_by` matches a shared account's `users.id` (via `SharedAccountRegistry.isSharedUserId`) → server unit test
- [ ] Session payload `isShared` is `false` for sessions whose `created_by` is a regular user's `users.id` → server unit test

### Boundary / contract

- [ ] **The set of shared-account user-ids is NOT exposed in any client-visible payload** (e.g., no `sharedAccountIds: string[]` on `/api/config`, no marker on user objects revealing membership in the shared-account set). The client must rely on the per-session `isShared` boolean instead. → schema test or boundary test
- [ ] Existing `Session.createdBy` field shape is preserved (no removal, no rename). Existing `useSessionFilter` `mine` filter continues to work unmodified. → typecheck + existing test
- [ ] Shared types in `packages/shared/src/types/session.ts` declare `isShared: boolean` (single source of truth for client/server). → typecheck
- [ ] Shared schema (Valibot) in `packages/shared/src/schemas/session.ts` validates `isShared: boolean`. → schema test
- [ ] `/api/config` response type is declared in shared types if a config-response type already exists, otherwise added there. → typecheck

### Edge cases (boundary values from design-principles)

- [ ] Empty session list: `/api/sessions` (or equivalent listing) returns `[]` without erroring on `isShared` derivation → server unit test
- [ ] Session whose `created_by` is `null` (legacy / unauthenticated): `isShared` is `false` (not `null`, not `undefined`) → server unit test

## Why each criterion matters

- **`sharedAccountsAvailable` gating**: prevents the client from rendering shared-session UI in single-user mode → consistent with design doc's opt-in stance ("Don't make it mandatory infrastructure").
- **`isShared` derivation server-side**: source of truth stays with `users` / configuration; future multi-account configurations can extend without client churn.
- **Shared-account set protection (boundary contract)**: design doc rejects exposing the *set* of shared-account user-ids to the client (this would let the client recover internal identity layout — "which users are shared accounts"). Per-session `isShared` is the safe abstraction.
- **`createdBy` preservation**: Multi-user UX (existing `mine` filter, future #683 tree view, future #684 owner display) requires per-session owner identification. The design doc's reject target is the *set* of shared-account ids, NOT the per-session `createdBy`.
- **Empty list / null `created_by` boundaries**: standard agent blind-spot per `design-principles.md` "Specify boundary values in design briefs".

## Notes for implementer

- Slice 1 (PR #764) landed `sessions.created_by`, `SharedAccountRegistry`, `serverConfig.AGENT_CONSOLE_SHARED_USERNAME`, and the upsert flow. This slice **consumes** those, not re-implements them.
- Cross-doc citation check: confirm `shared-orchestrator-session.md` §UI claims match the actual schema in `packages/server/src/database/migrations/v20*` and `packages/shared/src/schemas/session.ts`.
- Q1 (existing-mechanism check): existing `/api/config` route handles its own shape. Extend in place; do NOT add a sibling `/api/shared-config` endpoint.
- Q8 signature shape pre-estimate: if extending the shared `Session` type changes downstream call-site signatures, run `grep -rn "\bSession\b" packages/client/src packages/server/src --include="*.ts" --include="*.tsx" | grep -v __tests__ | wc -l` and report the count + affected file list in the PR description.

## Multi-PR Delivery (3-layer cleaning rule)

Per `core-responsibilities.md § Multi-PR Delivery`:

- PR title MUST be `feat: shared session UI affordance — server contract (Part 2a of #678) (closes #774)` — **no `closes #678` keyword for parent**.
- PR body MUST include `Closes #774` AND `Refs #678` — never `Closes #678`.
- Squash-merge commit message MUST be cleaned of any `closes #678` keyword.

## Correction history

The original AC (filed 2026-05-09) contained a misreading of the design doc. The Orchestrator wrote "`created_by` and the shared-account user-id list are NOT present in the session payload" as a single rule, conflating two separate things:

- The **set of shared-account user-ids** (which IS the design doc's reject target — leaks internal identity layout)
- The **per-session `createdBy`** (which is preserved — multi-user UX requires it)

The agent's pre-implementation Q1 check surfaced the conflict with existing `Session.createdBy` and `useSessionFilter` `mine` filter; the Orchestrator confirmed the misreading and revised the AC to its current form. The lesson — read what the design doc rejects as `(object, alternative)` pairs, not as standalone words — is being captured separately as feedback memory.

Refs #678

### PR DESCRIPTION
**Pull request by ms2sato**
## Summary

Server contract for the shared-session UI affordance (Part 2a of #678). Two additions to the data layer that the (future) Part 2b client UI will consume:

- **`/api/config.sharedAccountsAvailable: boolean`** — gate so the client can show / hide shared-session affordances based on whether `AGENT_CONSOLE_SHARED_USERNAME` resolves to a real OS account.
- **`Session.isShared: boolean`** (server-derived) — per-session signal computed via `SharedAccountRegistry.isSharedUserId(createdBy)`. Always a concrete boolean on the wire.

`Session.createdBy` is preserved as-is. The Issue's original AC misread the design doc's reject target (the *set* of shared-account user-ids — not per-session `createdBy`) as a single rule; the Issue body has been corrected, and this PR follows the corrected scope. Multi-user UX (`useSessionFilter` "mine" filter, future #683 tree view, future #684 owner display) continues to need per-session owner identification.

## Implementation notes

- `SessionConverterService.deriveIsShared(createdBy)` is the single writer (I-2). Both `toPublicSession` (in-memory) and `persistedToPublicSession` (paused, loaded from DB) call it.
- `SessionManagerOptions.sharedAccountLookup` is **optional** with a `() => false` default — so the dozens of unit-test SessionManager constructors don't need to thread the registry just to satisfy the converter dep. Production wires the real registry at `app-context.ts`.
- `ConfigResponse` is added to `packages/shared/src/types/auth.ts` (small + system-related; co-locates with `LoginResponse`/`CurrentUserResponse`). Client `lib/api.ts` re-exports it instead of duplicating.
- Boundary contract: shared-account user-id **set** is never exposed (`/api/config` test asserts no `sharedAccountIds` / `sharedAccounts` array leaks).

## Q8 signature shape pre-estimate

`grep -rn "\bSession\b" packages/client/src packages/server/src --include="*.ts" --include="*.tsx" | grep -v __tests__` = **318 references / 71 files**.

Change is **purely additive** (new required field on `SessionBase`). Production call sites that don't read `isShared` are unaffected. Test fixtures that build `Session` objects via literal needed the new field — 6 such fixtures updated in this PR.

## Architectural Invariants

- **I-2 (Single writer for derived values):** `deriveIsShared` in `SessionConverterService` is the only function computing `isShared`. All other code reads it from the `Session` object.
- **I-5 (Server as source of truth):** `isShared` is derived server-side and validated by Valibot on the wire (`SessionBaseSchema`). Client never computes it.
- **I-6 (Boundary validation):** `/api/config` response shape is declared in shared types (`ConfigResponse`) and exercised by tests asserting both presence of `sharedAccountsAvailable` and absence of any shared-account-id list. WebSocket session payloads are validated by `SessionBaseSchema` (4 new tests cover `isShared` true / false / non-boolean reject / missing reject).

## CodeRabbit findings

CodeRabbit CLI returned 1 `major` finding suggesting `isShared` be made optional (`isShared?: boolean`). **Rejected** — the AC explicitly requires `isShared` to be `false` (not `null`, not `undefined`) for legacy `created_by === null` sessions, and the server normalizes legacy DB rows through `deriveIsShared` in `persistedToPublicSession`. Wire contract is "always boolean", which is exactly the AC's intent. Making it optional would weaken the boundary and require every consumer to defensively coalesce.

## Verification

- ✅ `bun run typecheck` — exit 0
- ✅ `bun run test` — TEST_EXIT 0 (server 2513 pass / shared 324 / client 1361 / integration 29 / scripts+hooks+skills 362, 0 fail across 4527 tests)
- ✅ `bun run check:lang` — exit 0
- ✅ `node .claude/skills/orchestrator/preflight-check.js` — exit 0 (4 production files all covered)
- ✅ `coderabbit review --agent --base main` — 1 major finding, justified reject above; 0 CRITICAL / 0 HIGH (treating `major` as HIGH-equivalent and addressing via PR-body justification)

## Test plan

- [ ] CI green (typecheck + tests + preflight + language-lint)
- [ ] CodeRabbit GitHub-side review state
- [ ] Manual sanity (post-merge dogfood): `/api/config` shows `sharedAccountsAvailable: false` in default config; flipping `AGENT_CONSOLE_SHARED_USERNAME` to a real OS user flips it to `true`

Closes #774
Refs #678

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/server/src/routes/__tests__/api.test.ts [hunk 1]
```diff
@@ -2,8 +2,20 @@ import { describe, it, expect, beforeEach, afterEach } from 'bun:test';
...
+
+/** Minimal SystemCapabilitiesService mock that returns no detected capabilities. */
+function createMockSystemCapabilities(): SystemCapabilitiesService {
+  return {
+    detect: async () => {},
+    getCapabilities: () => ({ vscode: false }),
+    hasVSCode: () => false,
+    getVSCodeCommand: () => null,
+  } as unknown as SystemCapabilitiesService;
+}
```
[the remark this answers is not in the document]
Comment (ms2sato) [reply packages/server/src/routes/__tests__/api.test.ts:24]: Addressed in 25838e5. `createMockSystemCapabilities` now constructs a real `new SystemCapabilitiesService()` and seeds its private state via `Reflect.set` (same pattern as `__tests__/system.test.ts` to avoid the `which` shell-out). `fakeUserRepository as never` is replaced by a `satisfies UserRepository` literal — added the missing `findById` method (returns `null`, never invoked). No `as unknown as` / `as never` remaining.

File: packages/shared/src/schemas/__tests__/app-server-message.test.ts [hunk 2]
```diff
@@ -172,6 +174,43 @@ describe('AppServerMessageSchema', () => {
...
+        expect(output.session.isShared).toBe(false);
+      }
+    });
+
+    it('should reject session with non-boolean isShared', () => {
+      expectInvalid({
+        type: 'session-created',
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        session: { ...worktreeSession, isShared: 'true' as any },
+      });
```
[the remark this answers is not in the document]
Comment (ms2sato) [reply packages/shared/src/schemas/__tests__/app-server-message.test.ts:201]: Addressed in 25838e5. Removed `as any` and the eslint-disable comment per the suggestion. The spread `{ ...worktreeSession, isShared: 'true' }` widens the inferred type cleanly because `worktreeSession` is an unannotated `const` literal (no `boolean` constraint propagated), so no compile error — and `expectInvalid(unknown)` accepts the result.

**Comment by ms2sato** [comment]
## Acceptance check Q10 — Concerns Surfacing (acknowledged)

The orchestrator's acceptance walk surfaced 4 minor concerns. Owner judgment = acknowledge all + defer to follow-up Issues if/when triggered. Tracked in `memory/project_pending_triage_list.md` under "Sprint 2026-05-12 由来 (PR #778)" so they are not forgotten.

| # | Concern | Disposition |
|---|---|---|
| (a) | `SessionManagerOptions.sharedAccountLookup` optional fallback — silent test degradation risk if a test constructs SessionManager directly without specifying the lookup | Follow-up Issue to be filed when Slice 3+ tests intentionally exercise shared sessions; promote to required at that point. |
| (b) | `Session.isShared` has no production client reader yet (Slice 2b will consume) — interim dead-code-like state on the wire | Acknowledge; Slice 2b ([#775](https://github.com/ms2sato/agent-console/issues/775)) is dependency-tracked and scheduled next, low stall risk. |
| (c) | `/api/config` lacks Valibot wire-level validation (TS interface only) — matches existing pattern, not PR-attributable | Acknowledge; file a future Issue if a wrong-shape regression actually occurs. |
| (d) | Integration mock Partial-laxness in `review-queue-boundary.test.ts` — sibling of [#773](https://github.com/ms2sato/agent-console/issues/773) cast pattern | Bundle the codebase-wide sweep with #773 when it lands. |

Acceptance Q9 (glossary integrity) is also PARTIAL — `isShared` and `sharedAccountsAvailable` identifiers will be added to `docs/glossary.md` `### SharedSession` and `### Shared Account` entries via a follow-up commit on this PR before merge.
